solana/Dockerfile.wasm.m1

```diff
@@ -0,0 +1,90 @@
+# syntax=docker.io/docker/dockerfile:1.3@sha256:42399d4635eddd7a9b8a24be879d2f9a930d0ed040a61324cfdf59ef1357b3b2
+FROM docker.io/library/rust:1.49@sha256:a50165ea96983c21832578afb1c8c028674c965bc1ed43b607871b1f362e06a5 AS build
+
+RUN apt-get update && apt-get install -y libssl-dev libudev-dev pkg-config zlib1g-dev llvm clang
+RUN rustup component add rustfmt
+RUN rustup default nightly-2022-01-02
```

Comment:
We probably want this to match the toolchain we're using for all other solana code: nightly-2021-12-03